### DISCUSSION TITLE
Reservoir Life Cycle Query HIP-RA

### DISCUSSION THREAD

**Discussion by tb831637** [NatLabRockies/GEOPHIRES-X#263] [Q&A] [answered]
The input “reservoir life cycle” in HIP-RA is not clear as to the meaning. It has a very large effect on the output.

The impact of the variable “Reservoir Life Cycle” is huge and counterintuitive for me. I get less Producible Electricity as I enter a larger number for the Reservoir Life Cycle input……

**Comment by softwareengineerprogrammer** [comment]
@malcolm-dsider Can you comment on this?

**Comment by malcolm-dsider** [comment, marked as answer]
Starting it GitHub, I searched for “Reservoir Life Cycle,” and I see that it leads to a variable called `self.reservoir_life_cycle` in [hip_ra_x.py](https://github.com/NREL/GEOPHIRES-X/blob/28df988d6f004c400239383e1f48e2b4bed9adb8/src/hip_ra_x/hip_ra_x.py#L182-L194):

```
self.reservoir_life_cycle: Parameter = parameter_dict_entry(
            intParameter(
                'Reservoir Life Cycle',
                value=30,
                UnitType=Units.TIME,
                PreferredUnits=TimeUnit.YEAR,
                CurrentUnits=TimeUnit.YEAR,
                AllowableRange=list(range(1, 101, 1)),
                Required=True,
                ErrMessage='assume default Reservoir Life Cycle (30 years)',
                ToolTipText='Reservoir Life Cycle [30 years]',
            )
```

I then searched for “self.reservoir_life_cycle,” and find that it leads to just [one line of source code](https://github.com/NREL/GEOPHIRES-X/blob/28df988d6f004c400239383e1f48e2b4bed9adb8/src/hip_ra_x/hip_ra_x.py#L751):

```
maximum_power_kW = maximum_lifetime_electricity_kJ / (self.reservoir_life_cycle.value * 365 * 24 * 3600)
```

So that tells me how it is used.

It is useful to know what `maximum_lifetime_electricity_kJ` represents, so I search on that, and I find [this](https://github.com/NREL/GEOPHIRES-X/blob/28df988d6f004c400239383e1f48e2b4bed9adb8/src/hip_ra_x/hip_ra_x.py#L734):

```
maximum_lifetime_electricity_kJ = amount_fluid_produced_kg * fluid_exergy_kJ_per_kg
```

So `maximum_lifetime_electricity_kJ` represents the total amount of electricity that can be produced during the lifetime of the project, based on the amount of fluid in the reservoir, and the exergy in that fluid. This calculator makes the assumption that you will extract all that available energy in that reservoir within the lifetime of the project. If you make the lifetime longer, the per year number looks worse. If you shorten it, the per year number looks better.

---

*Note: formatting lightly edited with code links and for readability by @softwareengineerprogrammer*

**Reply by tb831637** [reply]
maximum_lifetime_electricity_kJ is not a standard output. Can I get that as a standard or some such?

**Reply by malcolm-dsider** [reply]
Could you clarify, please? Are you asking for this number to appear in the final report? Or are you asking us to do some unit conversion on it?

It is worth noting that in reviewing to code, this number is modified by the conversion efficiency before it is used.

It is also worth noting that the total amount of energy is based on the total amount of fluid you can produce, and this calculation assumes you can produce 100% of the fluid, like wringing a sponge dry.

It is also worth noting that this is based upon the assumption you are producing a fluid but not replacing it and reheating it again.

And this whole premise is moot if you are talking about conduction-only closed loops. No fluids produced at all in that case.

**Reply by tb831637** [reply]
This is good context as to boundary conditions. My question at the root was to try to find a way to relate HIP and Geophires results. As you say, it is not useful to do so. No need to do anything.

**Comment by malcolm-dsider** [comment]
You responded via email with:

I found all that and will post up as you ask. My question was not worded well. I am working on a prediction of the size of the prize for a project along the Texas GC. One piece of a larger area shows ~20MW electricity via Geophires. The larger areal extent would equate to ~400MW proportionally I would assume just working the area differences…..

Maybe I don’t understand the intent of HIP-RA. I was thinking that I could use it to show a similar value via a different means. The impact of the variable “Reservoir Life Cycle” is huge and counterintuitive for me. I get less Producible Electricity as I enter larger number for the Reservoir Life Cycle input……

**Comment by malcolm-dsider** [comment]
This is working as expected - it is not possible, nor is it intended, that you can get to a similar value using different means (GEOPHIRES vs HIP-RA). I think of HIP-RA as a top-down assessment that calculates all the possible heat energy that it is present in the entire reservoir, and GEOPHIRES is a bottom-up calculation of what is possible to be recovered; the two are always going to be far apart, as it is not possible to recover anywhere near 100% of the heat in the entire reservoir. In your case, the recovery seems to be about 5% (20 MS vs 400 MW).

**Reply by tb831637** [reply]
So if I input the full areal scope of the development (# of wells) into Geophires, I get an answer of ~400MW continuous power. 

When I input the full scope parameters needed for HIP-RA, I get ~40MW producible electricity (reservoir) for a 30y life cycle and ~1GW for 1 year..... It appears that HIP-RA is showing an average production of that 1GW over the course of the life cycle and the 1-year value is the total possible from the top-down assessment view.

**Reply by malcolm-dsider** [reply]
Yes, I would say that is correct, 1-year is the total possible (but not practical).
